docs/build/autogenerate.rst

```diff
@@ -405,18 +407,24 @@ is set to True::
 .. note::
 
    The default type comparison logic (which is end-user extensible) currently
-   works for **major changes in type only**, such as between ``Numeric`` and
-   ``String``.     The logic will **not** detect changes such as:
+   works by comparing the generated SQL for a column. It does this in two steps-
 
-   * changes between types that have the same "type affinity", such as
-     between ``VARCHAR`` and ``TEXT``, or ``FLOAT`` and ``NUMERIC``
 
-   * changes between the arguments within the type, such as the lengths of
-     strings, precision values for numerics, the elements inside of an
-     enumeration.
+   * First, it compares the outer type of each column such as ``VARCHAR``
+     or ``TEXT``. Dialect implementations can have synonyms that are considered
+     equivalent- this is because some databases support types by converting them
+     to another type. For example, in ``NUMERIC`` and ``DECIMAL`` are considered
+     equivalent.
 
-   Detection of these kinds of parameters is a long term project on the
-   SQLAlchemy side.
+   * Next, the arguments within the type, such as the lengths of
+     strings, precision values for numerics, the elements inside of an
+     enumeration are compared. If BOTH columns have arguments AND they are
+     different, a change will be detected. If one column is jsut set to the
+     default and the other has arguments, we don't currently detect this. The
```

Comment:
**mike bayer** (zzzeek) wrote:

"and the other has arguments, Alembic will pass on attempting to compare these.  The rationale is that it is difficult to detect what a database backend sets as a default value without generating false positives".